### PR TITLE
Refacto `ModuleManager::uninstall`

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Refato `ModuleManager::uninstall`.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | `php bin/console prestashop:module uninstall your_module_name`.
| Fixed ticket?     | N/A
| Related PRs       | N/A
| Sponsor company   | N/A

### CODE REVIEW

**Comment by Hlavtox** [comment]
@PululuK Please update the PR description. What does it improve? Why was the original code wrong?

File: src/Core/Module/ModuleManager.php [hunk 1]
```diff
@@ -156,12 +156,12 @@ public function uninstall(string $name, bool $deleteFiles = false): bool
...
         $uninstalled = $module->onUninstall();
 
-        if ($deleteFiles && $path = $this->moduleRepository->getModulePath($name)) {
-            $this->filesystem->remove($path);
-        }
-
         $this->dispatch(ModuleManagementEvent::UNINSTALL, $module);
 
+        if ($deleteFiles) {
+            return $uninstalled && $this->delete($name);
```
Comment (matthieu-rolland) [inline src/Core/Module/ModuleManager.php:162]: Shouldn't this be noted as a BC Break? In the case where `$this->delete()` doesn't return true, the result of the `uninstall` function won't be the same before and after this PR.

(Which is fine, BC breaks are allowed, this is PS 9)

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by kpodemski** [comment]
@PululuK I know you're busy, but with PR's like this is hard to understand the full context :-( any info?

**Comment by kpodemski** [comment]
Since we have had no news from you for more than 20 days, I'm closing this PR. Feel free to re-open it if you find the motivation to finish it. Don't hesitate to ping me if you have any questions or doubts.

**Pull request closed without merging**